pipit/util/test_generator.py

```diff
@@ -0,0 +1,366 @@
+import random
+import textwrap
+import pandas as pd
+
+
+class FakeNode:
+    """
+    A single node of the calling tree used to produce fake traces.
+    Represents a single function call.
+    """
+
+    def __init__(
+        self, name, exc_time, mpi_type="", mpi_tgt=0, mpi_volume=0, mpi_time=0
+    ):
+        self.name = name
+        self.exc_time = exc_time
+        self.inc_time = exc_time
+        self.children = {}  # run_time -> child node
+        self.is_mpi = mpi_type != ""
+        self.mpi_type = mpi_type
+        self.mpi_tgt = mpi_tgt
+        self.mpi_volume = mpi_volume
+        self.mpi_time = mpi_time
+        self.total_nodes = 1
+        self.parent = None
+
+    def grow_inc_time(self, time):
```

Comment:
I'm assuming that a lot of these are "internal" functions. Unless there is a use case that you see, I don't think we should expose these and make them "private" by prepending them with double underscores.

Reply:
Good point. I'll make this and the other code changes tomorrow.